torchseg/losses/_functional.py

```diff
@@ -151,32 +151,37 @@ def soft_jaccard_score(
 ) -> torch.Tensor:
     assert output.size() == target.size()
     if dims is not None:
-        intersection = torch.sum(output * target, dim=dims)
-        cardinality = torch.sum(output + target, dim=dims)
+        intersection = torch.sum(output * target, dim = dims)
+        cardinality = torch.sum(output + target, dim = dims)
     else:
         intersection = torch.sum(output * target)
         cardinality = torch.sum(output + target)
 
     union = cardinality - intersection
-    jaccard_score = (intersection + smooth) / (union + smooth).clamp_min(eps)
+    jaccard_score = (intersection + smooth) / (union + eps)
     return jaccard_score
 
 
 def soft_dice_score(
     output: torch.Tensor,
     target: torch.Tensor,
+    power: float =  1.0,
     smooth: float = 0.0,
     eps: float = 1e-7,
-    dims=None,
+    dims = None,
 ) -> torch.Tensor:
     assert output.size() == target.size()
     if dims is not None:
-        intersection = torch.sum(output * target, dim=dims)
-        cardinality = torch.sum(output + target, dim=dims)
+        intersection = torch.sum(output * target, dim = dims)
+        output_pow = torch.sum(output ** power, dim = dims)
+        target_pow = torch.sum(target ** power, dim = dims)
+        cardinality = output_pow + target_pow
     else:
         intersection = torch.sum(output * target)
-        cardinality = torch.sum(output + target)
-    dice_score = (2.0 * intersection + smooth) / (cardinality + smooth).clamp_min(eps)
+        output_pow = torch.sum(output ** power)
+        target_pow = torch.sum(target ** power)
+        cardinality = output_pow + target_pow
+    dice_score = (2.0 * intersection + smooth) / (cardinality + eps)
```

Comment:
Do we not need to add smooth to cardinality?

Reply:
eps and smooth basically do the same thing (smooth avoid a value of 0, eps avoid nan). Supposing that smooth should be very small (same order as eps), we could just use one of the two, but we can also treat them separately, let me know

i think leaving them both gives more customization